src/components/Auth/SignUp/SignUp.js

```diff
@@ -184,71 +184,71 @@ class SignUp extends Component {
 
   handleSubmit = event => {
     event.preventDefault();
+    this.setState({
+      signupErrorMessage: '',
+    });
+
     const {
       email,
       password,
-      emailError,
-      passwordConfirmError,
+      emailErrorMessage,
+      passwordConfirmErrorMessage,
       isCaptchaVerified,
     } = this.state;
+
     const { openSnackBar } = this.props;
-    const { getSignup } = this.props.actions;
 
     if (!isCaptchaVerified) {
       this.setState({
         captchaVerifyErrorMessage: 'Please verify that you are a human.',
       });
     }
 
-    if (!emailError && !passwordConfirmError && isCaptchaVerified) {
+    if (
+      !emailErrorMessage &&
+      !passwordConfirmErrorMessage &&
+      isCaptchaVerified
+    ) {
       this.setState({ loading: true });
-      getSignup({
-        email,
-        password: encodeURIComponent(password),
-      })
+      actions
+        .getSignup({
+          email,
+          password: encodeURIComponent(password),
+        })
         .then(({ payload }) => {
           if (payload.accepted) {
-            this.setState(
-              {
-                success: true,
-                loading: false,
-              },
-              () => {
-                openSnackBar({ snackBarMessage: payload.message });
-                this.closeDialog();
-              },
-            );
+            this.setState({
+              password: '',
+              confirmPassword: '',
+              passwordStrength: '',
+              passwordScore: -1,
+              signupErrorMessage: payload.message,
+              success: true,
+              loading: false,
+            });
           } else {
-            this.setState(
-              {
-                password: '',
-                confirmPassword: '',
-                passwordScore: -1,
-                passwordStrength: '',
-                success: false,
-                loading: false,
-              },
-              () => {
-                openSnackBar({ snackBarMessage: 'Failed. Try Again' });
-              },
-            );
+            this.setState({
+              password: '',
+              success: false,
+              loading: false,
+            });
+            openSnackBar({
+              snackBarMessage: 'Signup Failed. Try Again',
+              snackBarDuration: 6000,
```

Comment:
No need to add 6000 duration, it is set to default 4000

Reply:
@fragm3 Removed that! Please review the changes again